### PR TITLE
Bugfix/powerup reset on replay

### PR DESCRIPTION
**Pull request by mikeslgoh**
- reset player rotation on restart
- powerups do not carry over when game is restarted (either from end game or pause panel)
- car hats are back in the player selection

### CODE REVIEW

**Comment by mikeslgoh** [comment]
[Powerup follow over to next game if click replay](https://trello.com/c/h5EwY1Ue/597-powerup-follow-over-to-next-game-if-click-replay)

File: Assets/_PartyParkade/Script/Game/PlayerController.cs [hunk 1]
```diff
@@ -57,6 +57,15 @@ public void ResetPlayer(Vector3 spawnPoint)
         {
             Car.SetCarPosition(spawnPoint);
             SetPointText(0);
+            ResetInventory();
+        }
+
+        private void ResetInventory()
+        {
+            if (m_inventory)
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: does this work on C#? it's doing a null check right?

Comment (mikeslgoh) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: Yeah I'm doing a null check. Not sure if this should work for C#. Should I do != null to be safe?

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: You can do some research on this, i'm trusting the compiler, but u should do some research on this.

Comment (mikeslgoh) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: https://forum.unity.com/threads/optimizing-null-check-null-vs-bool-out-functions.482118/

Near the end of the thread, there is apparently a method called ReferenceEquals(UnityObject, null) that might be better in terms of performance as well

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: interesting, that's good to know. 
But this isn't called all the time, but we know this work right?

Comment (mikeslgoh) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: yup it works

Comment (Gursaanj) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: This definitely works!

Comment (Gursaanj) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: Most people just do the != null for readability so no one confuses the variable for a bool, from what I understand

Comment (m4rtint) [reply Assets/_PartyParkade/Script/Game/PlayerController.cs:65]: ah that's fine, but good to know C# accepts it.

**Pull request merged**